classes/controller/AdminController.php

```diff
@@ -2521,7 +2521,7 @@ public function renderForm()
             $helper->tpl_vars = $this->getTemplateFormVars();
             $helper->show_cancel_button = (isset($this->show_form_cancel_button)) ? $this->show_form_cancel_button : ($this->display == 'add' || $this->display == 'edit');
 
-            $back = urldecode(Tools::getValue('back', ''));
+            $back = urldecode(str_replace('+', '%2B', Tools::getValue('back', '')));
```

Comment:
Should it be easier to use rawurlencode instead?

Reply:
Definitely 🤦‍♂️